[chore]: Update stack resolver to support ghc 9.4.8

worm2fed opened the pull request.

worm2fed committed:
deps: update stack resolver to support ghc 9.4.8

worm2fed committed:
deps: drop allow-newer flag

gzh commented:
> @gzh do not like that `template-haskell` is restricted at `stack.yaml` here..

stack.yaml does not matter for anything except development of the package itself. but okay, there was no specific intent to limit TH, and allow-newer is not necessary.

worm2fed commented:
@gzh `template-haskell` restriction goes from `syb-with-class`

worm2fed committed:
deps: switch from `ixset` to `ixset-typed`

Reason:
- `ixset` depends on old template haskell
- `ixset` can lead to runtime exceptions

worm2fed commented:
@gzh I can see that LTS for GHC 9.6.3 has been released recently, and it includes `ixset-typed` (only)

<img width="735" alt="image" src="https://github.com/gzh/haskell-wamp/assets/6579275/e868ed81-6d1e-40e3-b18d-1250f991dedb">
